utils/update_arch_timings.py

```diff
@@ -90,6 +112,39 @@ def find_timings(timings, bel, location, site, bels, corner, speed_type):
        the selected `bel` in `location` and `site`. If timings
        are not found, empty dict is returned"""
 
+    def get_timing(cell, delay, corner, speed_type):
+        """
+        Gets timing for a particular cornet case. If not fount then chooses
+        the next best one.
+        """
+        entries = cell[delay]['delay_paths'][corner.lower()]
+        entry = entries.get(speed_type, None)
+
+        if speed_type == 'min':
+            if entry is None:
+                entry = entries.get('avg', None)
+            if entry is None:
+                entry = entries.get('max', None)
+
+        elif speed_type == 'avg':
+            if entry is None:
+                entry = entries.get('max', None)
+            if entry is None:
+                entry = entries.get('min', None)
+
+        elif speed_type == 'max':
+            if entry is None:
+                entry = entries.get('avg', None)
+            if entry is None:
+                entry = entries.get('min', None)
+
+        if entry is None:
+            # if we failed with desired corner, try the opposite
```

Comment:
This is invalid, and almost never the right thing to do?

Reply:
With that code I targeted a situation when you have eg. only the slow corner case given in SDF and the fast one is requested. Since there is no other data the code returns what's available.

This is meant to be able to use SDF with incomplete timing specs.

Reply:
> This is meant to be able to use SDF with incomplete timing specs.

When would this ever be true?

Reply:
This happens for Quicklogic devices when we have SDFs that do not specify all the speed corners which makes the import script crash.

Reply:
Is it that the data doesn't exist?  Or the SDF's are incomplete?

Reply:
To clarify, the underlying hardware definitely has a FF and SS corner, and the data from those corners is required to properly model both setup and hold constraints over PVT.  If the part cannot exhibit hold violations, it might be better to just not emit the opposite corner, rather than subing another corners data.